shakmap input

schmidni opened the pull request:
By passing oq.base_path to the parsing functions, I can now make sure that the files can be found independently from where the "oq engine" command is run.

I have also changed the tests to vary the input format for the files in order to have all the cases (relative paths, absolute paths and zipped files) covered.

micheles commented:
Are you sure that we need to pass `base_path` everywhere? `readinput.get_input_files` is there to ensure that relative file names are expanded to absolute path names so there should be not need to pass `base_path` around.

schmidni commented:
My bad, I did not look at the zip function properly and was not aware of the problems.

``get_input_files`` still has some problems with the shakemap files. I will put some work into it and see that I can use it for the zip AND shakemap case.

schmidni commented:
``readinput.get_input_files`` now parses all possible shakemap file paths, adds the correct files to the ``fnames`` list and expands them in oqparams.

additionally I updated the tests and added two new ones for the zip feature.

micheles commented on openquake/commonlib/readinput.py:
```diff
@@ -1160,16 +1160,18 @@ def get_input_files(oqparam, hazard=False):
+        # local files
+        for key, val in uri.items():
+            if key == 'fname' or key.endswith('_url'):
+                val = val.replace('file://', '').replace('file:', '')
```
I understand 'file://` but not `file:`. I thought `file:` without the `//` was an error.

schmidni replied:
Yeah, its a bit confusing. 

I removed the reference to the file:// solution from the documentation now. Absolute Paths can also be specified now without the "file:". file: denotes still a valid URI therefore I left it in.

- A valid file URI must therefore begin with either file:/path (no hostname), file:///path (empty hostname), or file://hostname/path. If host is omitted, it is taken to be "localhost", the machine from which the URL is being interpreted.

Admittedly my solution is not working when using a Hostname, then again, that would be going a bit far imo.

micheles replied:
I would support only `file:///path`

schmidni replied:
ok

micheles commented on openquake/hazardlib/shakemap/parsers.py:
```diff
@@ -160,16 +160,31 @@ def get_array_usgs_xml(kind, grid_url, uncertainty_url=None):
+                            'uncertainty.xml') as f2:
+                        return get_shakemap_array(f1, f2)
+                except FileNotFoundError:
+                    pass
```
Perhaps a warning saying that the grid file is missing?

schmidni replied:
If leaving out the engine doesn't even go that far since already oqvalidation.is_valid_shakemap returns an error.
 
```ValueError: Expected parameters ['kind', 'grid_url', 'uncertainty_url'] in shakemap_uri, got ['kind', 'grid_url']```

The case where no uncertainty grid is specified only happens in "tests" at the moment.

schmidni replied:
yes, will fix it, just saw that there is another problem, let me fix that quickly

schmidni replied:
I added the warning. Also it was not possible to have 
```
shakemap_uri = {
     "kind": "usgs_xml",
     "grid_url": "shakemap_files/grid.zip"}
```
Because oqvalidation raised an error. Fixed that too now.

micheles commented:
Apart from the minor comments, LGTM
